### PR TITLE
Add workflow for Jest and Vitest testing

### LINKED ISSUE #90: Add new workflow for Vitest testing
Need to create a new workflow that will perform our unit testing for us.

This will allow us to have streamlined unit testing, as well as make sure that all the code that is written will not negatively impact previously written code.

### PR DESCRIPTION
**Pull request by tbmcallister**
## Description
This will allow us to have streamlined unit testing, as well as make sure that all the code that is written will not negatively impact previously written code. It will automatically run our Jest testing, as well as coverage reports.

This PR performs both Vitest and Jest testing, so it covers the client and server side testing

## Related Issue
This PR closes #90 and #92

## Type of Changes
Please select the relevant option:
- [ ] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Documentation update
- [x] Other (please specify)
      
This is a workflow addition(not sure what that falls under).

## Things to Update
Please add any features, buttons, or pages that might need updating in the future:
- [ ] Button Functunality 
- [ ] Code Documentation 
- [ ] Mobile Responsiveness 
- [x] Code testing 
- [ ] Other (please specify)

## Screenshots 
image

image

## Additional Notes
Currently, the testing is very limited, so new testing will be needed to be added in future PRs

### CODE REVIEW

File: .github/workflows/jest.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Jest
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '24 21 * * 4'
```
Comment (KingJordan152) [inline .github/workflows/jest.yml:10]: What is the cron job for?

**Review by KingJordan152 — DISMISSED** [review]
Overall this workflow looks good. You correctly install Jest and run both the `test` and `coverage` commands. This workflow should be extremely useful for us as we start integrating more unit tests. Keep up the great work!

**Review by KingJordan152 — COMMENTED** [review]
Please refer to the review I left on #91 and let me know if my suggestion is applicable. If it is, then I'd recommend implementing it. Otherwise, let me know and I'll approve what you have.

File: .github/workflows/jest.yml (hunk 1 above)
Comment (tbmcallister) [reply .github/workflows/jest.yml]: it was leftover from the github actions default job, this has been removed

**Comment by tbmcallister** [comment]
This PR is a combination of the Vitest and the Jest testing PRs, so we only need to have one testing.yml file.

**Review by KingJordan152 — APPROVED** [review]
Thank you for implementing my suggestion! Approved ✅

**Review by Linkaloo — APPROVED** [review]

**Review by Aligary — APPROVED** [review]

**Pull request merged**